### PR TITLE
#12 - grade columns and grade crud

### LINKED ISSUE #12: Grades management
This could be implemented as separate view or as a part of #11. I would like to have a grid with:
- [ ] list of all students in group in rows (sorting would be cool!)
- [ ] list of columns defined for every group (maybe columns should have types like "nullable boolean", "classic grade", "text", "final grade"?)

Current solution:

obraz

### PR DESCRIPTION
**Pull request by kamilpiech97**
This should close #12.
Added CRUD for managing grade columns in student group and adding grades for students.
Added some tests.

Preview:

[Screencast from 22-10-23 11:08:16.webm](https://github.com/blumilksoftware/keating/assets/29115024/7067b811-f21e-4b99-879f-bb0c1cfb1c2f)

### CODE REVIEW

File: resources/js/Pages/Dashboard/CourseSemester/Grade/Index.vue [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+                >
+                  <ArrowRightIcon class="w-6 hover:text-gray-500" />
+                </div>
+              </div>
+            </TableHeader>
+          </template>
+          <template #body>
+            <TableRow v-for="student in students.data" :key="student.id">
+              <TableCell>
+                {{ student.index_number }} ({{ student.first_name }} {{ student.surname }} )
```
Comment (EwelinaSkrzypacz) [inline resources/js/Pages/Dashboard/CourseSemester/Grade/Index.vue:215]:
```suggestion
                {{ student.index_number }} ({{ student.first_name }} {{ student.surname }})
```

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. If we're checking for presence, it should remain as it currently is. But once the cell has a state (presence), any state changes should be triggered by a double click. Why? Because now we can't write negative assessment on red background.

2. When I write too long string on grades, I see error:

image

3. We should change delete column icon from trash to x mark, because we have x mark icon in whole application.

**Review by EwelinaSkrzypacz — COMMENTED** [review]
I believe that the width of the columns should be fairly consistent. Alternatively, should we consider adjusting it based on the length of the contained text?

image

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. Icons is too small - maybe instead of arrows we should make draggable columns>

image

2. What a shame that we have to scroll the table, maybe we can do this something with it?

**Review by EwelinaSkrzypacz — COMMENTED** [review]
This is a nitpicking a bit, but I think I will change some colors:
- instead of `bg-green-300` please set `bg-lime-200`
- change `bg-red-300` to `bg-rose-300`
- set text color to `text-grey-900`

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Review by KarolZygadlo — APPROVED** [review]

**Pull request merged**